### PR TITLE
Fix withdrawables amount

### PR DESCRIPTION
**Pull request by maikelordaz**
Related to git issue #160
I add more test than the ones on the issue due to some changes I made to track better the participants who were expelled before being beneficiary.

<h2>Issue tracking expelled ones</h2>
Currently the code set the mapping `isBeneficiary[address]` to `true` for every one on the corresponding cycle, this was making it harder to track on the tests the ones that were expelled before their cycle.

<h2>Main changes on the facets</h2>
<h3>Fund Facet</h3>
- On `_awardBeneficiary` function if the beneficiary is already expelled the mpping `isBeneficiary[address]` is not set to true.</br>
- On the `_freezePot` function there is now a check if is expelled and is not a beneficiary, the money pot is not frozen.</br>
- On the `withdrawFund` function there is now a check if is expelled and is not a beneficiary, it can withdraw the moneymoney pot, there is also a require to check if the cycle is the one where the expelled one should be beneficiary.</br>

<h3>Getters Facet</h3>
- Fix the error on `getWithdrawableUserBalance` where I used `msg.sender` instead of `user`</br>

<h2>Tests</h2>
- When the collateral state is AcceptingCollateral, there is no withdrawable amount.</br>
- When the collateral state is ReleasingCollateral, and the term expires, withdrawable should be equal to locked balance, can withdraw collateral.</br>
- When the collateral state is ReleasingCollateral, and the term finish, withdrawable should be equal to locked balance, can withdraw collateral.</br>
- When the term is on going and somebody is expelled before being beneficiary, before the cycle expelled one should be beneficiary, withdrawable collateral equal to locked balance, can withdraw collateral.</br>
- When the term is on going and somebody is expelled before being beneficiary, the cycle expelled one should be beneficiary, withdrawable collateral equal to locked balance, can withdraw fund, can withdraw collateral,  money pot not frozen.</br>
- When the term is on going and somebody is expelled before being beneficiary, cycle after the one expelled one should be beneficiary, withdrawable collateral equal to locked balance, can withdraw fund, can withdraw collateral.</br>
- When the term finish and somebody is expelled before being beneficiary, can withdraw collateral, can withdraw fund.</br>

### CODE REVIEW

**Comment by maikelordaz** [comment]
Another solution that I think could be cleaner is to write a new mapping on the `Fund` struct, something like `mapping(address => bool) isExpelledBeforeBeneficiary` and track that one, and fetch it on `getParticipantFundSummary` function, but that will change the contract's ABI

**Comment by MohammedTaka** [comment]
isExpelledBeforeBeneficiary is a good idea. Added a Jira task: https://takadao.atlassian.net/browse/TT-1464

File: contracts/facets/FundFacet.sol [hunk 1]
```diff
@@ -250,12 +250,37 @@ contract FundFacet is IFund {
...
+                    unchecked {
+                        ++i;
+                    }
+
+                    continue;
+                }
+                cycleShouldBeBeneficiary = i + 1;
+                break;
+            }
+            require(
```
Comment (MohammedTaka) [inline contracts/facets/FundFacet.sol]: No need for this check, before the expelled before beneficiary's cycle, his fund.beneficiariesPool is 0 until his cycle. The whole if-statement is not needed by extention

File: contracts/facets/FundFacet.sol [hunk 2]
```diff
@@ -348,14 +373,16 @@ contract FundFacet is IFund {
...
-        if (EnumerableSet.remove(_fund._participants, beneficiary)) {
-            // ..Then add them to the benificiaries set
-            EnumerableSet.add(_fund._beneficiaries, beneficiary);
-        } // If this if-statement fails, this means we're dealing with a graced defaulter
+        if (!IGetters(address(this)).wasExpelled(_term.termId, beneficiary)) {
+            // Remove participant from participants set..
+            if (EnumerableSet.remove(_fund._participants, beneficiary)) {
+                // ..Then add them to the benificiaries set
+                EnumerableSet.add(_fund._beneficiaries, beneficiary);
+            } // If this if-statement fails, this means we're dealing with a graced defaulter
```
Comment (MohammedTaka) [inline contracts/facets/FundFacet.sol]: chore: Please remove this comment, this is invalid info

// If this if-statement fails, this means we're dealing with a graced defaulter

**Review by MohammedTaka — COMMENTED** [review]
Small changes but can be merged afterwards

File: contracts/facets/FundFacet.sol (hunk 2 above)
Comment (maikelordaz) [reply contracts/facets/FundFacet.sol]: Commit 7f52c88

File: contracts/facets/FundFacet.sol (hunk 1 above)
Comment (maikelordaz) [reply contracts/facets/FundFacet.sol]: Commits 2c1903f and 818311d

**Comment by maikelordaz** [comment]
> isExpelledBeforeBeneficiary is a good idea. Added a Jira task: https://takadao.atlassian.net/browse/TT-1464

I added this on commits 3ea23c9 and bc02966, on commit b95edbd fixed the tests according this.

I wrote a new getter function only to track this mapping (to avoid problems on frontend or backend), but I'll write a report of some functions that can be combined, but that will change the ABI. For example this new getter and `isBeneficiary` can be deleted and use `getUserFundSummary` to track all mappings related to the user.

Now with this getter we can track who is expelled before being beneficiary, and on his cycle the user can be set as beneficiary without affecting anything else. 

Every test passes

@MohammedTaka

**Comment by maikelordaz** [comment]
@MohammedTaka I made the unit tests requested commits 8aff6e1 and 4fe8056

**Review by MohammedTaka — APPROVED** [review]

**Pull request merged**